nowcast/tests/sensorization/test_sensor.py

```diff
@@ -0,0 +1,91 @@
+        test_sensors = [SensorConfig("a", "b", "c", 1), SensorConfig("x", "y", "z", 2)]
+        test_ground_truth_sensor = SensorConfig("i", "j", "k", 3)
+        test_ground_truth = [LocationSeries("ca", "state")]
+        assert compute_sensors(
+            date(2020, 5, 5), test_sensors, test_ground_truth_sensor, test_ground_truth, False
+        ) == {
+            SensorConfig("i", "j", "k", 3): [LocationSeries("ca", "state", [date(2020, 5, 1)], [1.5])],
+            SensorConfig("a", "b", "c", 1): [LocationSeries("ca", "state", [date(2020, 5, 4)], [2.5])],
+        }
+
+
+class TestHisoricalSensors:
+
+    @patch("delphi_nowcast.sensorization.sensor.get_historical_sensor_data")
+    def test_historical_sensors_some_data(self, historical):
+        """Test non empty data is returned for first two sensors."""
+        historical.side_effect = [(LocationSeries(dates=[date(2020, 1, 1)], values=[2]), []),
+                                  (LocationSeries(dates=[date(2020, 1, 3)], values=[4]), []),
+                                  (LocationSeries(), [])]
+        test_sensors = [SensorConfig("i", "j", "k", 3), SensorConfig("a", "b", "c", 1), SensorConfig("x", "y", "z", 2)]
+        test_ground_truth = [LocationSeries("ca", "state")]
+        assert historical_sensors(
+            None, None, test_sensors, test_ground_truth) == {
+            SensorConfig("i", "j", "k", 3): [LocationSeries(dates=[date(2020, 1, 1)], values=[2])],
+            SensorConfig("a", "b", "c", 1): [LocationSeries(dates=[date(2020, 1, 3)], values=[4])]
+        }
+
+    @patch("delphi_nowcast.sensorization.sensor.get_historical_sensor_data")
+    def test_historical_sensors_no_data(self, historical):
+        """Test nothing returned for any sensor."""
+        historical.return_value = (LocationSeries(), [])
+        test_sensors = [SensorConfig("i", "j", "k", 3), SensorConfig("a", "b", "c", 1), SensorConfig("x", "y", "z", 2)]
+        test_ground_truth = [LocationSeries("ca", "state")]
+        assert historical_sensors(
+            None, None, test_sensors, test_ground_truth) == {}
+
+
+class TestExportToCSV:
+
+    def test__export_to_csv(self):
+        """Test export creates the right file and right contents."""
+        test_sensor = SensorConfig(source="src",
+                                   signal="sig",
+                                   name="test",
+                                   lag=4)
+        test_value = LocationSeries("ca", "state", [date(2020, 1, 1)], [1.5])
+        with tempfile.TemporaryDirectory() as tmpdir:
+            out_files = _export_to_csv(test_value, test_sensor, date(2020, 1, 5), receiving_dir=tmpdir)
+            assert len(out_files) == 1
+            out_file = out_files[0]
+            assert os.path.isfile(out_file)
+            assert out_file.endswith("issue_20200105/src/20200101_state_sig.csv")
+            with open(out_file) as f:
+                assert f.read() == "sensor_name,geo_value,value\ntest,ca,1.5\n"
```

Comment:
This is fine for the file size now, but it might be more future-proof and set a better precedent for other tests to read the file using `csvreader` or `pandas` and comparing against a Python object of appropriate type rather than the string version.

Reply:
updated to use pandas